app/controllers/country_controller.rb

```diff
@@ -65,33 +66,45 @@ def protected_areas
 
   private
 
+  def get_number_of_national_desinitions
+    number_of_national_desinitions = 0
+    designations_list = @stats_data[:wdpa][:designations][:designations]
+    designations_list.each do |designation|
+      total = designation[:total]
+      number_of_national_desinitions = total if designation[:type] == 'National' && total.is_a?(Integer)
+    end
+    number_of_national_desinitions
+  end
```

Comment:
Just a suggestion, I dont have idea if the designation_list is quite long or not. But it can keep iterating even if the National type is found. Here, you can try this one. It will find the National and then breakout and return total and if not found then it returns 0.

def get_number_of_national_designations
    @stats_data[:wdpa][:designations][:designations].find do |designation|
        designation[:type] == 'National' && designation[:total].is_a?(Integer)
    end&.fetch(:total, 0)
end